OMEdit/OMEditLIB/Modeling/ModelWidgetContainer.cpp

```diff
@@ -235,6 +235,35 @@ GraphicsView::GraphicsView(StringHandler::ViewType viewType, ModelWidget *pModel
   mAllItems.clear();
 }
 
+GraphicsView::~GraphicsView()
+{
+  /* When the scene is deleted it will delete all the items inside it.
+   * We need to delete the items that are not part of the scene.
+   */
+  foreach (Element *pElement, mOutOfSceneElementsList) {
+    if (pElement->getOriginItem()) {
+      delete pElement->getOriginItem();
+    }
```

Comment:
I guess this means that the origin item will still leak if an `Element` isn't added to a `GraphicsView`, though I'm not sure what their relation is.

One possible solution to that would be to remove `getOriginItem` and instead have e.g. `addToScene/removeFromScene`-methods in `Element` that takes a `GraphicsView` and calls `addItem/removeItem` on the origin item. That way you can keep track of if someone else "owns" the item, and otherwise delete it in `~Element`.

Reply:
Yes, but what about elements that are never actually added to it in the first place? I'm not sure those exist in practice though.

Reply:
Such elements exists but not in the new instance api approach.